NullPointerException

I'm trying to make a build test but i get that error:
```
[DEBUG] Configuring mojo com.github.gino0631:pkg-maven-plugin:1.1:product from plugin realm ClassRealm[plugin>com.github.gino0631:pkg-maven-plugin:1.1, parent: sun.misc.Launcher$AppClassLoader@10dea4e]
[DEBUG] Configuring mojo 'com.github.gino0631:pkg-maven-plugin:1.1:product' with basic configurator -->
[DEBUG]   (f) buildDirectory = F:\DigitalMediaServer\target
[DEBUG]   (f) alignment = bottomLeft
[DEBUG]   (f) scaling = proportional
[DEBUG]   (s)  = F:\DigitalMediaServer/src/main/resources/images/splash.png
[DEBUG]   (f) resources = [com.github.gino0631.pkg.maven.Resource@90265a]
[DEBUG]   (f) background = com.github.gino0631.pkg.maven.Background@32191e
[DEBUG]   (s)  = F:\DigitalMediaServer/src/main/external-resources/osx/license.rtf
[DEBUG]   (f) file = F:\DigitalMediaServer/src/main/external-resources/osx/gpl-2.0-fr.rtf
[DEBUG]   (f) lang = fr
[DEBUG]   (f) resources = [com.github.gino0631.pkg.maven.Resource@1accff0, com.github.gino0631.pkg.maven.Resource@559a69]
[DEBUG]   (f) license = com.github.gino0631.pkg.maven.License@5583d4
[DEBUG]   (s)  = F:\DigitalMediaServer/src/main/external-resources/readme.rtf
[DEBUG]   (f) readme = com.github.gino0631.pkg.maven.Readme@185217c
[DEBUG]   (s)  = F:\DigitalMediaServer/src/main/external-resources/conclusion.rtf
[DEBUG]   (f) conclusion = com.github.gino0631.pkg.maven.Conclusion@1b202ff
[DEBUG]   (f) distribution = com.github.gino0631.pkg.maven.Distribution@a84a12
[DEBUG]   (f) outputFile = DigitalMediaServer-0.0.2-SNAPSHOT.pkg
[DEBUG]   (f) packageIdentifier = org.digitalmediaserver.dms.pkg
[DEBUG]   (f) packageTitle = Digital Media Server
[DEBUG]   (f) packageVersion = 0.0.2-SNAPSHOT
[DEBUG]   (f) includes = [**/*.sh]
[DEBUG]   (f) excludes = []
[DEBUG]   (f) fileMode = 493
[DEBUG]   (f) permissionSets = [com.github.gino0631.pkg.maven.PermissionSet@cc02bb]
[DEBUG]   (f) root = F:\DigitalMediaServer\target\dms-macosx-x64
[DEBUG] -- end configuration --
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:56 min
[INFO] Finished at: 2018-05-24T00:52:14+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.gino0631:pkg-maven-plugin:1.1:product (create-macosx-pkg) on project dms: Execution create-macosx-pkg of goal com.github.gino0631:pkg-maven-plugin:1.1:product failed.: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.github.gino0631:pkg-maven-plugin:1.1:product (create-macosx-pkg) on project dms: Execution create-macosx-pkg of goal com.github.gino0631:pkg-maven-plugin:1.1:product failed.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
```

I suspect that i'm doing something wrong so any help is welcome ;)
Here is the my _pom.xml_:
```
                         <build>
				<plugins>
					<plugin>
						<groupId>org.apache.maven.plugins</groupId>
						<artifactId>maven-antrun-plugin</artifactId>
						<executions>
							<execution>
								<id>create-dir-macos</id>
								<phase>process-resources</phase>
								<goals>
									<goal>run</goal>
								</goals>
								<configuration>
									<target>
										<mkdir dir="${project.build.directory}/${project.artifactId}-macosx-x64" />
										<chmod dir="${project.build.directory}/${project.artifactId}-macosx-x64" type="dir" perm="0755"/>
									</target>
								</configuration>
							</execution>
						</executions>
					</plugin>
					<plugin>
						<groupId>com.github.gino0631</groupId>
						<artifactId>pkg-maven-plugin</artifactId>
						<version>1.1</version>
						<executions>
							<execution>
								<id>create-macosx-pkg</id>
								<phase>package</phase>
								<goals>
									<goal>product</goal>
								</goals>
								<configuration>
									<root>${project.build.directory}/${project.artifactId}-macosx-x64</root>
									<outputFile>${project.name.camel}-${project.version}.pkg</outputFile>
									<packageIdentifier>${project.groupId}.${project.artifactId}.pkg</packageIdentifier>
									<packageVersion>${project.version}</packageVersion>
									<packageTitle>${project.name}</packageTitle>
									<distribution>
										<background>
											<alignment>bottomLeft</alignment>
											<scaling>proportional</scaling>
											<resources>
												<resource>${project.basedir}/src/main/resources/images/splash.png</resource>
											</resources>
										</background>
										<license>
											<resources>
												<resource>${project.basedir}/src/main/external-resources/osx/license.rtf</resource>
												<resource>
													<file>${project.basedir}/src/main/external-resources/osx/gpl-2.0-fr.rtf</file>
													<lang>fr</lang>
												</resource>
											</resources>
										</license>
										<readme>${project.basedir}/src/main/external-resources/readme.rtf</readme>
										<conclusion>${project.basedir}/src/main/external-resources/conclusion.rtf</conclusion>
									</distribution>
									<permissionSets>
										<permissionSet>
											<includes>
												<include>**/*.sh</include>
											</includes>
											<excludes />
											<fileMode>0755</fileMode>
											<directoryMode />
											<uid />
											<gid />
										</permissionSet>
									</permissionSets>
									<!--
									<signing>
										<keystore>${project.basedir}/../dms.p12</keystore>
										<storetype>PKCS12</storetype>
										<storepass>${project.name.camel}</storepass>
										<tsa>http://timestamp.apple.com/ts01</tsa>
										<skip>false</skip>
									</signing>
									-->
								</configuration>
							</execution>
						</executions>
					</plugin>
				</plugins>
			</build>
```

Oups, sorry, i should have looked at the log before posting it...i guess that was probably because i had the _Distribution_ file open in Notepad++
```
[ERROR] Failed to execute goal com.github.gino0631:pkg-maven-plugin:1.1:product (create-macosx-pkg) on project dms: Execution create-macosx-pkg of goal com.github.gino0631:pkg-maven-plugin:1.1:product failed.: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.github.gino0631:pkg-maven-plugin:1.1:product (create-macosx-pkg) on project dms: Execution create-macosx-pkg of goal com.github.gino0631:pkg-maven-plugin:1.1:product failed.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.NullPointerException
    at java.util.Objects.requireNonNull (Objects.java:203)
    at com.github.gino0631.xar.impl.XarBuilderImpl.setSigning (XarBuilderImpl.java:154)
    at com.github.gino0631.pkg.ProductBuilder.build (ProductBuilder.java:238)
    at com.github.gino0631.pkg.maven.ProductMojo.execute (ProductMojo.java:106)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
```
[pkg-debug2.txt](https://github.com/gino0631/pkg/files/2035280/pkg-debug2.txt)

If it come from the Payload missing a more meaningful error message should have been displayed. (as it generate a `installKBytes=0`)
Just an idea though.
I'm a total macOS noobs and as a non english native i had difficulties to understand what you said about the Payload and if anything special need to be done to handle it.

--- a/pkg-core/src/main/java/com/github/gino0631/pkg/ProductBuilder.java
+++ b/pkg-core/src/main/java/com/github/gino0631/pkg/ProductBuilder.java
@@ -235,7 +235,9 @@ public final class ProductBuilder {
 
         // Build XAR
         try (XarBuilder xarBuilder = XarBuilder.getInstance()) {
-            xarBuilder.setSigning(signingPrivateKey, signingCertificates, signingProvider, signingTsa);
+            if (signingPrivateKey != null && signingCertificates != null) {
+                xarBuilder.setSigning(signingPrivateKey, signingCertificates, signingProvider, signingTsa);
+            }
 
             Files.walkFileTree(flatDir, new SimpleFileVisitor<Path>() {
                 Deque<XarBuilder.Container> xarPath = new LinkedList<>();

--- a/pkg-core/src/main/java/com/github/gino0631/pkg/jbomutils/MkBom.java
+++ b/pkg-core/src/main/java/com/github/gino0631/pkg/jbomutils/MkBom.java
@@ -66,7 +66,7 @@ public class MkBom {
                 n.gid = Integer.parseInt(elements[2]);
                 n.size = 0;
                 n.checksum = 0;
-                //n.linkNameLength = 0;              
+                //n.linkNameLength = 0;
                 if ((n.mode & 0xF000) == 0x4000) {
                     n.type = TNodeType.KDirectoryNode;
                     n.linkName = "";
@@ -241,6 +241,7 @@ public class MkBom {
             }
 
             if (num_paths > 1) {
+                root_paths.indices.add(new BomPathIndices());
                 root_paths.indices.get(current_path).index0 = (new BomPaths(Tools.getBIS(bom.getBlock(last_paths_id)))).forward = bom.addBlock(Tools.getBytes(paths), current_path_size);
                 root_paths.indices.get(current_path).index1 = last_file_info;
                 tree.child = bom.addBlock(Tools.getBytes(root_paths), path_size);

